File: test-db.js
const admin = require('firebase-admin');

// Initialize admin
const serviceAccount = require('./backend/service-account.json');
if (!admin.apps.length) {
    admin.initializeApp({ credential: admin.credential.cert(serviceAccount) });
}
const db = admin.firestore();

async function test() {
    // Current UTC date
    const date = new Date().toISOString().split('T')[0];
    const docRef = db.collection('quant_predictions').doc(date);
    const snap = await docRef.get();
    
    if (!snap.exists) {
        console.log(`No quant_predictions found for ${date}`);
        // Let's check yesterday
        const yest = new Date(Date.now() - 86400000).toISOString().split('T')[0];
        const ySnap = await db.collection('quant_predictions').doc(yest).get();
        if (ySnap.exists) {
            console.log(`Found yesterday ${yest}`);
            const data = ySnap.data();
            console.log(JSON.stringify(data.predictions.slice(0, 2), null, 2));
        }
    } else {
        const data = snap.data();
        console.log(`Found today ${date}`);
        console.log(JSON.stringify(data.predictions.slice(0, 2), null, 2));
    }
}

test().then(() => process.exit(0)).catch(console.error);

File: test_db.py
import sys
sys.path.insert(0, "./backend/quant")
from quant_pipeline import _init_firebase
import firebase_admin
from firebase_admin import firestore
import json
from datetime import datetime, timezone, timedelta

def run():
    if not _init_firebase():
        print("Firebase init failed")
        return
        
    db = firestore.client()
    dates = [(datetime.now(timezone.utc) - timedelta(days=i)).strftime("%Y-%m-%d") for i in range(5)]
    
    for date in dates:
        doc = db.collection("quant_predictions").document(date).get()
        if doc.exists:
            print(f"--- {date} ---")
            data = doc.to_dict()
            preds = data.get("predictions", [])
            if preds:
                print(json.dumps(preds[0], indent=2))
                return
    print("No predictions found in last 5 days")

run()
